policy/release/sbom_cyclonedx.rego

```diff
@@ -119,6 +119,36 @@ deny contains result if {
 	result := lib.result_helper(rego.metadata.chain(), [component.purl, property.name, msg])
 }
 
+# METADATA
+# title: Disallowed package external references
+# description: >-
+#   Confirm the CycloneDX SBOM contains only packages without disallowed
+#   external references. By default all external references are allowed. Use the
+#   "disallowed_external_references" rule data key to provide a list of type-pattern pairs
+#   that forbid the use of an external reference of the given type where the reference url
+#   matches the given pattern.
+# custom:
+#   short_name: disallowed_package_external_references
+#   failure_msg: Package %s has reference %q of type %q which is disallowed%s
+#   solution: Update the image to not use a package with a disallowed external reference.
+#   collections:
+#   - redhat
+#   - policy_data
+#   effective_on: 2024-07-31T00:00:00Z
+deny contains result if {
+	some s in sbom.cyclonedx_sboms
+	some component in s.components
+	some reference in component.externalReferences
+	some disallowed in lib.rule_data(_rule_data_external_references_key)
+
+	reference.type == disallowed.type
+	regex.match(object.get(disallowed, "url", ""), object.get(reference, "url", ""))
+
+	msg := regex.replace(object.get(reference, "value", ""), `(.+)`, ` to "$1"`)
```

Comment:
Would be nice to have the `msg` generation covered in a test.

Reply:
Thanks for pointing this out. I think `value` is never defined there.

I updated it in [1bc7ca9](https://github.com/enterprise-contract/ec-policies/pull/1060/commits/1bc7ca9d2c59d665f38435a52ea7716af8ad7500) to be more useful.